plugins/flytekit-vscode/flytekitplugins/vscode/decorator.py

```diff
@@ -148,26 +225,19 @@ def inner_wrapper(*args, **kwargs):
             download_vscode(
                 code_server_remote_path=code_server_remote_path,
                 code_server_dir_name=code_server_dir_name,
+                plugins_remote_paths=plugins_remote_paths
             )
 
             # 2. Launches and monitors the VSCode server.
             # Run the function in the background
-            logger.info(f"Start the server for {server_up_seconds} seconds...")
             child_process = multiprocessing.Process(
                 target=execute_command, kwargs={"cmd": f"code-server --bind-addr 0.0.0.0:{port} --auth none"}
             )
 
             child_process.start()
-            time.sleep(server_up_seconds)
+            exit_handler(max_idle_seconds, child_process, post_execute)
 
-            # 3. Terminates the server after server_up_seconds
-            logger.info(f"{server_up_seconds} seconds passed. Terminating...")
-            if post_execute is not None:
-                post_execute()
-                logger.info("Post execute function executed successfully!")
-            child_process.terminate()
-            child_process.join()
-            sys.exit(0)
+            return
```

Comment:
nit: we don't need to have return here since the program will end in exit_handler

Reply:
Done thanks